containers/hawkular-openshift-security-filter/src/main/java/org/hawkular/openshift/auth/org/hawkular/openshift/namespace/NamespaceOverrideMapper.java

```diff
@@ -0,0 +1,279 @@
+            }
+        }
+
+        Cluster cluster = clusterBuilder.build();
+        cluster.init();
+
+        Session session = cluster.connect();
+
+        return session;
+    }
+
+    /**
+     * Creates a keyspace if it does not already exist
+     * @param keyspace The name of the keyspace
+     */
+    private void ensureKeyspace(String keyspace) {
+        ResultSet resultset = session.execute("SELECT * FROM system_schema.keyspaces WHERE keyspace_name = '" + keyspace + "';");
+        if (!resultset.iterator().hasNext()) {
+            session.execute("CREATE KEYSPACE " + keyspace + " WITH replication = {'class': 'SimpleStrategy', 'replication_factor': '1'}  AND durable_writes = true;");
+        }
+    }
+
+    private boolean hasMappingTable() {
+        ResultSet resultset = session.execute("SELECT table_name FROM system_schema.tables WHERE keyspace_name='openshift_metrics' AND table_name='metrics_mappings';");
+        return resultset.iterator().hasNext();
+    }
+
+    private void createMappingTable() {
+        if (!hasMappingTable()) {
+            session.execute("CREATE TABLE openshift_metrics.metrics_mappings (project_id text PRIMARY KEY, project_name text);");
+        }
+    }
+
+    private Map<String,String> getMappingTable() {
+        Map<String,String> mappings = new HashMap<>();
+        if (hasMappingTable()) {
+            ResultSet resultSet = session.execute("SELECT project_id,project_name FROM openshift_metrics.metrics_mappings;");
+
+            Iterator<Row> ri = resultSet.iterator();
+            while (ri.hasNext()) {
+                Row row = ri.next();
+                mappings.put(row.getString("project_id"), row.getString("project_name"));
+            }
+
+        }
+
+        return mappings;
+    }
+
+    private void writeMappingTable(Map<String,String> mappings) {
+        createMappingTable();
+
+        StringBuilder builder = new StringBuilder();
+        builder.append("BEGIN BATCH ");
+
+        for (Map.Entry<String,String> entry: mappings.entrySet()) {
+            builder.append(" INSERT INTO openshift_metrics.metrics_mappings (project_id, project_name) VALUES('"+ entry.getKey() + "','" + entry.getValue() + "');");
+        }
+        builder.append(" APPLY BATCH;");
+
+        session.execute(builder.toString());
+    }
+
+    private Set<String> getTenants() {
+        Set<String> tenants = new HashSet<>();
+        ResultSet resultset = session.execute("SELECT * FROM system_schema.keyspaces WHERE keyspace_name = '" + "hawkular_metrics" + "';");
+        if (!resultset.iterator().hasNext()) {
+            return tenants;
+        }
+
+        try {
+            // An invalid query exception will occur if the table does not exists.
+            // If the table does not exist, then no tenants have been stored yet so we just return the empty tenant set.
+            ResultSet resultSet = session.execute("SELECT DISTINCT tenant_id,type from hawkular_metrics.metrics_idx;");
```

Comment:
Unlike the other query which I said should be avoided, this query while it appears very similar, is often ok. You are only querying the partition key which is indexed and fast to query as it should not require touching SSTables.